### ISSUE TITLE
How to run with android studio?

### ISSUE THREAD

**Issue by MohammadGh2179** [SagerNet/SagerNet#643]
Hello, I want to open and run this project with android studio and make some changes, then build an apk.
Is this possible?

Os : Ubuntu

![image](https://user-images.githubusercontent.com/42517494/220086224-257d3210-1475-45c4-9a2c-665c583ddb99.png)

Please help me

**Comment by rJoel01** [comment]
Yes u can! once the application runs and works well on the simulator you can yust press on Build "on top left of the screen", and then press Build(APK).
Android Studio will build the apk, when it's done click "locate" on the bottom right. (the apk file you want is the .apk)
 
Hope i helped u :)
